src/components/Icon/Icon.test.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { ThemeProvider } from 'styled-components/native';
+
+import theme from '../../common/themeSelectors/theme/mock-theme.json';
+import {
+  Icon,
+  IconProps,
+  IconColors,
+} from './Icon';
+
+const renderIcon = (fn, props: Omit<IconProps, 'theme'>) => (fn(
+  <ThemeProvider theme={theme}>
+    <Icon {...props} />
+  </ThemeProvider>,
+));
+
+const sampleNames = [
+  'filled-action-add',
+  'outlined-action-micrec',
+  'outlined-content-wellness',
+  'outlined-finance-bank',
+];
+
+describe('Icon component', () => {
+  it('should render component with all possible icons', () => {
+    const props = { color: 'default' as IconColors };
+
+    sampleNames.forEach((name) => {
```

Comment:
Sugestão: aqui poderíamos usar o `it.each` do `jest` em vez do `forEach`.
Fica mais aderente à API do Jest e, de quebra, torna a listagem de testes mais abrangente.

```tsx
describe('Icon component', () => {
  const props = { color: 'default' as IconColors };

  it.each(sampleNames)("should render correctly when icon is %p", (name) => {
    const wrapper = renderIcon(renderer.create, { name, ...props });
    expect(wrapper.toJSON()).toMatchSnapshot(`Icon component - name: ${name}`);
  });

)};
```